types/rdf-dataset-indexed/dataset.d.ts

```diff
@@ -0,0 +1,39 @@
+import * as RDF from 'rdf-js';
+import { Readable } from 'stream';
+
+declare class DatasetIndexed<Q extends RDF.Quad = RDF.Quad> implements RDF.Dataset<Q> {
```

Comment:
`InQuad` should be `BaseQuad`. But more importantly, it only implements `DatasetCore` doesn't it? There are some differences, hence https://github.com/rdf-ext/rdf-dataset-indexed/pull/3.

Might be best to extract (most of) this change.

Reply:
> InQuad should be BaseQuad.

👍 

> But more importantly, it only implements DatasetCore doesn't it?

TBH, I'm not sure. The implementation is similar to that of `rdf-ext` and the API surface does look fairly compatible with `Dataset` 🤷‍♂

Reply:
They're not all implemented (eg `toCanonical`, `union`). I want to say I found some with different signatures too, but was a while ago when I tried to use it.

Reply:
Do you propose to go back to `DatasetCore` and keep it that way until rdf-ext/rdf-dataset-indexed#3 is merged?

Reply:
Yeah. Can keep the methods though, as long as they match what it actually implements.

Reply:
Shoot, you're right. The methods are not exactly the same. Let me revisit that

Reply:
Better to do these in a separate PR IMO (unrelated changes).